### PR TITLE
Adding nvidia-hpc-sdk based on upstream PR

### PR DESCRIPTION
**Pull request by pramodk**
- based on new PR upstream https://github.com/spack/spack/pull/19139

### CODE REVIEW

**Comment by pramodk** [comment]
retest this please

File: var/spack/repos/builtin/packages/nvidia-hpc-sdk/package.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    foundation to the CPU and through the interconnect. It is the only
+    comprehensive, integrated SDK for programming accelerated computing
+    systems."""
+
+    homepage = "https://developer.nvidia.com/hpc-sdk"
+
+    maintainers = ['Nischay-Pro']
+
+    for ver, packages in _versions.items():
+        key = "{0}_{1}".format(platform.system(), platform.machine())
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/nvidia-hpc-sdk/package.py]: So… no cross-compiling :)

Comment (pramodk) [reply var/spack/repos/builtin/packages/nvidia-hpc-sdk/package.py]: yes, no cross compiling but you just have to by the node with target arch :D

**Comment by pramodk** [comment]
Retest this please

**Comment by pramodk** [comment]
Retest this please

File: lib/spack/spack/compilers/nvhpc.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.compiler import Compiler
+
+
+class Nvhpc(Compiler):
```
Comment (matz-e) [inline lib/spack/spack/compilers/nvhpc.py:9]: Is this from upstream?

**Review by matz-e — DISMISSED** [review]
Seems good to me, I guess the hold-up will be MPI?

File: lib/spack/spack/compilers/nvhpc.py (hunk 2 above)
Comment (pramodk) [reply lib/spack/spack/compilers/nvhpc.py:9]: yup!

**Review by ohm314 — DISMISSED** [review]
lgtm

**Review by matz-e — DISMISSED** [review]

**Comment by pramodk** [comment]
> Seems good to me, I guess the hold-up will be MPI?

I assume you are referring to `-isystem` flag. We are using FindMPI in CoreNEURON and not depend on mpicc/mpicxx wrappers. So that allows us to continue using PGI.

**Review by matz-e — APPROVED** [review]

**Pull request merged**